src/tmpfiles/tmpfiles.c

```diff
@@ -2821,10 +2819,12 @@ int main(int argc, char *argv[]) {
 
         mac_selinux_finish();
 
-        if (r < 0)
+        if (r < 0 || r2 == -ENOMEM)
                 return EXIT_FAILURE;
         else if (invalid_config)
                 return EX_DATAERR;
+        else if (r2 < 0)
```

Comment:
maybe find a more expressive name for "r2"? "ret_process" or so, to indicate that this is about the return value fo the processing calls?